metadata-ingestion/src/datahub/utilities/registries/domain_registry.py

```diff
@@ -0,0 +1,43 @@
+import logging
+from typing import List, Optional
+
+from datahub.ingestion.graph.client import DataHubGraph
+
+logger = logging.getLogger(__name__)
+
+
+class DomainRegistry:
+    """A class that makes it easy to resolve domains using DataHub"""
+
+    def __init__(
+        self,
+        cached_domains: Optional[List[str]] = [],
+        graph: Optional[DataHubGraph] = None,
+    ):
+        self.domain_registry = {}
+        if cached_domains:
+            # isolate the domains that don't seem fully specified
+            domains_needing_resolution = [
+                d
+                for d in cached_domains
+                if (not d.startswith("urn:li:domain") and d.count("-") != 4)
```

Comment:
You can create custom domains on Datahub with urn like:
`urn:li:domain:engineering` in this case you don't need to resolve the urn as well.
I think it is ok to say we will break that functionality to only specify the uuid in the config.
I mean the real uuid will work but I guess the friendly uuid get broken after this change.

Reply:
I think we can solve for this with a second query to test if the urn-based domain exists for this bare name. Will do that in a follow-up PR this week.